src/annoylib.h

```diff
@@ -138,19 +192,11 @@ struct Angular {
       return random.flip();
   }
   static inline void create_split(const vector<Node*>& nodes, int f, Random& random, Node* n) {
-    // Sample two random points from the set of nodes
-    // Calculate the hyperplane equidistant from them
-    size_t count = nodes.size();
-    size_t i = random.index(count);
-    size_t j = random.index(count-1);
-    j += (j >= i); // ensure that i != j
-    T* iv = nodes[i]->v;
-    T* jv = nodes[j]->v;
-    T i_norm = get_norm(iv, f);
-    T j_norm = get_norm(jv, f);
+    static vector<T> best_iv(f, 0), best_jv(f, 0);
+    two_means<S, T, Random, Angular<S, T, Random> >(nodes, f, random, true, &best_iv[0], &best_jv[0]);
```

Comment:
not sure, i can try. when i work with templates i really just add/remove "typename" and add/remove template arguments randomly until it compiles :)